apps/api/src/app/topics/use-cases/create-topic-subscribers/create-topic-subscribers.use-case.ts

```diff
@@ -0,0 +1,39 @@
+import { TopicSubscribersEntity, TopicSubscribersRepository } from '@novu/dal';
+import { ConflictException, Injectable } from '@nestjs/common';
+
+import { CreateTopicSubscribersCommand } from './create-topic-subscribers.command';
+
+import { TopicSubscribersDto } from '../../dtos/topic-subscribers.dto';
+
+@Injectable()
+export class CreateTopicSubscribersUseCase {
+  constructor(private topicSubscribersRepository: TopicSubscribersRepository) {}
+
+  async execute(command: CreateTopicSubscribersCommand) {
+    const entity = this.mapToEntity(command);
+
+    const topicSubscribers = await this.topicSubscribersRepository.create(entity);
+
+    return this.mapFromEntity(topicSubscribers);
+  }
+
+  private mapToEntity(domainEntity: CreateTopicSubscribersCommand): TopicSubscribersEntity {
+    return {
+      _environmentId: TopicSubscribersRepository.convertStringToObjectId(domainEntity.environmentId),
+      _organizationId: TopicSubscribersRepository.convertStringToObjectId(domainEntity.organizationId),
+      _topicId: TopicSubscribersRepository.convertStringToObjectId(domainEntity.topicId),
+      _userId: TopicSubscribersRepository.convertStringToObjectId(domainEntity.userId),
+      subscribers: [],
+    };
+  }
+
+  private mapFromEntity(topicSubscribers: TopicSubscribersEntity): TopicSubscribersDto {
+    return {
+      ...topicSubscribers,
+      _environmentId: TopicSubscribersRepository.convertObjectIdToString(topicSubscribers._environmentId),
+      _organizationId: TopicSubscribersRepository.convertObjectIdToString(topicSubscribers._organizationId),
+      _topicId: TopicSubscribersRepository.convertObjectIdToString(topicSubscribers._topicId),
+      _userId: TopicSubscribersRepository.convertObjectIdToString(topicSubscribers._userId),
+    };
+  }
```

Comment:
Do you mind bringing it up for the next tech summit? Want to discuss it with the team to make sure we have a unified way of doing it across the code :)